packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -870,23 +870,22 @@ class FileCollection extends DocumentCollection {
 
     let { contentType, contentLength, checksum, lastModifiedDate, ifMatch } =
       options || {}
+
     if (!contentType) {
-      if (isBuffer) {
-        contentType = CONTENT_TYPE_OCTET_STREAM
-      } else if (isFile) {
-        contentType =
-          data.type ||
-          getFileTypeFromName(data.name.toLowerCase()) ||
-          CONTENT_TYPE_OCTET_STREAM
-        if (!lastModifiedDate) {
-          lastModifiedDate = data.lastModifiedDate
-        }
-      } else if (isBlob) {
-        contentType = data.type || CONTENT_TYPE_OCTET_STREAM
-      } else if (isStream) {
-        contentType = CONTENT_TYPE_OCTET_STREAM
-      } else if (typeof data === 'string') {
+      if (typeof data === 'string') {
         contentType = 'text/plain'
+      } else {
+        if (data.type) {
+          // The type is specified in the file object
+          contentType = data.type
+        } else {
+          // Extract the name from the path and infer the type
+          const sPath = path.split('?')
+          const params = sPath.length > 1 ? sPath[1] : ''
+          const name = new URLSearchParams(params).get('Name')
+          contentType =
+            getFileTypeFromName(name.toLowerCase()) || CONTENT_TYPE_OCTET_STREAM
+        }
       }
     }
```

Comment:
it miss the `lastModifiedDated` code in your new version, no? https://github.com/cozy/cozy-client/pull/1027/files#diff-76c9574e3b7f8c2ec150af6ff717b3d38ff269adb500ea34697860b335e5ef79L881-L883

Reply:
And since there is no test for it, we should add one ^^'

Reply:
My bad, I missed that. Fair enough for the test

Reply:
There it is: 5e3158e